crates/api/src/space.rs

```diff
@@ -0,0 +1,82 @@
+//! Kitsune2 space related types.
+
+use crate::*;
+use std::sync::Arc;
+
+/// Handler for events coming out of Kitsune2 such as messages from peers.
+pub trait SpaceHandler: 'static + Send + Sync + std::fmt::Debug {
+    /// We have received an incomming message from a remote peer.
+    ///
+    /// If this callback handler returns an `Err` response, the connection
+    /// will be closed immediately.
+    //
+    // Note: this is the minimal low-level messaging unit. We can decide
+    //       later if we want to handle request/response tracking in
+    //       kitsune itself as a convenience or if users of this lib
+    //       should have to implement that if they want it.
+    fn incoming_message(
+        &self,
+        peer: AgentId,
+        data: bytes::Bytes,
```

Comment:
Should this not be typed? Like an enum for the messages Kitsune supports?

Reply:
I think that decoding the bytes and then matching against kitsune-own message types for further processing will be done in this function.

Reply:
The `data` here is the opaque payload of something like receive_remote_signal. It's for holochain_p2p to apply any deserialization beyond that including picking the encoding scheme (although we know that will be messagepack).

Separately we still need to decide if kitsune is going to internally do request/response matching as a convenience to holochain, or if we'd prefer to move that complexity up a level to keep kitsune simple. But, I figured we can defer that decision a bit until we know if kitsune's internal modules are going to need that support anyways.